cityofaustin/atd-moped
# Refactor user edit form

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/9697

Updates:
1. Separates new user and edit user actions from `StaffForm` and adds more config props so the form uses what is handed to it
2. Componentizes the activate/inactivate button to isolate the activate/inactivate API calls
3. Componentizes the Staff form error modal to hopefully help debug https://github.com/cityofaustin/atd-data-tech/issues/10085
4. Componentizes the Staff form confirm modal

## Testing
**URL to test:** 
https://md-9697-user-edit-form--atd-moped-main.netlify.app/moped/

**Steps to test:**
1. Go to https://md-9697-refactor-user-edit-form--atd-moped-main.netlify.app/moped/staff, log in as an admin, and create a new user (I used `mike.dilley+editform@austintexas.gov`)
2. Edit your new user's details like role, first name, last name, title, workgroup, or password.
3. Inactivate your user
4. Activate your user with a new password (all fields besides password should be disabled)
5. Inactivate your user to cleanup the staging Cognito user pool and the staging DynamoDB table

---
#### Ship list
